utils/cmut__utils__compile_resource.cmake

```diff
@@ -0,0 +1,139 @@
+
+
+include("${CMAKE_CURRENT_LIST_DIR}/../cmut_message.cmake")
+
+
+function(cmut__utils__get_variable_name VARIABLE_NAME INPUT_FILE)
+    # get the filename and replace all ' ' and '.' with '_'
+    get_filename_component(SOURCE_FILE_NAME ${INPUT_FILE} NAME)
+    string(REGEX REPLACE "[^a-zA-Z0-9\-]" "_" SOURCE_FILE_NAME_VARIABLE "${SOURCE_FILE_NAME}")
+    # create Name ouf output files .h & .cpp
+    set(${VARIABLE_NAME} "${SOURCE_FILE_NAME_VARIABLE}" PARENT_SCOPE)
+endfunction()
+
+function(cmut__utils__get_header_name FILE_NAME INPUT_FILE OUTPUT_FOLDER)
+    cmut__utils__get_variable_name(SOURCE_FILE_NAME_VARIABLE "${INPUT_FILE}")
+    # create Name ouf output files .h & .cpp
+    if ("${OUTPUT_FOLDER}" STREQUAL "")
+        set(${FILE_NAME} "GENREATED__${SOURCE_FILE_NAME_VARIABLE}.h" PARENT_SCOPE)
+    else ()
+        set(${FILE_NAME} "${OUTPUT_FOLDER}/GENREATED__${SOURCE_FILE_NAME_VARIABLE}.h" PARENT_SCOPE)
+    endif ()
+endfunction()
+
+function(cmut__utils__get_source_name FILE_NAME INPUT_FILE OUTPUT_FOLDER)
+    cmut__utils__get_variable_name(SOURCE_FILE_NAME_VARIABLE "${INPUT_FILE}")
+    # create Name ouf output files .h & .cpp
+    if ("${OUTPUT_FOLDER}" STREQUAL "")
+        set(${FILE_NAME} "GENREATED__${SOURCE_FILE_NAME_VARIABLE}.cpp" PARENT_SCOPE)
+    else ()
+        set(${FILE_NAME} "${OUTPUT_FOLDER}/GENREATED__${SOURCE_FILE_NAME_VARIABLE}.cpp" PARENT_SCOPE)
+    endif ()
+endfunction()
+
+
+function(cmut__utils__get_header_lock LOCK_VALUE INPUT_FILE)
+    cmut__utils__get_variable_name(SOURCE_FILE_NAME_VARIABLE "${INPUT_FILE}")
+    # create The header multiple include lock
+    string(TOUPPER "GENREATED__${SOURCE_FILE_NAME_VARIABLE}.h" HEADER_LOCK)
+    string(REGEX REPLACE "[^A-Z0-9]" "_" HEADER_LOCK "${HEADER_LOCK}")
+    # create Name ouf output files .h & .cpp
+    set(${LOCK_VALUE} "${HEADER_LOCK}" PARENT_SCOPE)
+endfunction()
+
+
+function(cmut__utils__create_header_from_file INPUT_FILE OUTPUT_FOLDER)
+    # Check if the input file exist:
+    if ( NOT DEFINED INPUT_FILE )
+        message( FATAL_ERROR "cmut__utils__create_header_from_file: File does not exist: '${INPUT_FILE}'" )
+    endif()
+    cmut_info(" cmut__utils__create_header_from_file: File: '${INPUT_FILE}'" )
+    # Read the data form the input file
+    file(READ "${INPUT_FILE}" contents HEX)
+    get_filename_component(SOURCE_FILE_NAME ${INPUT_FILE} NAME)
+    cmut__utils__get_variable_name(SOURCE_FILE_NAME_VARIABLE "${INPUT_FILE}")
+    cmut__utils__get_header_name(FILE_HEADER ${INPUT_FILE} ${OUTPUT_FOLDER})
+    cmut__utils__get_source_name(FILE_SOURCE ${INPUT_FILE} ${OUTPUT_FOLDER})
+    cmut__utils__get_header_lock(HEADER_LOCK ${INPUT_FILE})
```

Comment:
header guard store in 
/usr/include/project_name/foo/my_file.h
should be
project_name__foo__my_file_h

Reply:
done